### PR TITLE
feat(docker-compose): yaml parser option `removeTemplates`

### PR DESCRIPTION
**Pull request by Nemental**
## Changes

Add option `removeTemplates: true` to docker-compose managers `parseSingleYaml` function call.
## Context

My docker compose files are templated with jinja2 because I use them with ansible for my deployment automation.
Things like that aren't possible:
```yaml
...
services:
  app:
    image: collabora/code:24.04.6.1.1
    restart: unless-stopped
    environment:
      {{ services['app']['environment'] | to_nice_yaml | indent(6) | trim }}
    networks:
      - traefik
      - collabora
...
```
```
"stack": "YAMLException: missed comma between flow collection entries (7:18)

 4 |     image: collabora/code:24.04.6.1.1
 5 |     restart: unless-stopped
 6 |     environment:
 7 |       {{ services['app']['environment'] | to_nice ...
----------------------^
 8 |     networks:
 9 |       - traefik
    at generateError (./node_modules/js-yaml/lib/loader.js:183:10)
    at throwError (./node_modules/js-yaml/lib/loader.js:187:9)
    at readFlowCollection (./node_modules/js-yaml/lib/loader.js:758:7)
    at composeNode (./node_modules/js-yaml/lib/loader.js:1442:11)
    at readFlowCollection (./node_modules/js-yaml/lib/loader.js:780:5)
    at composeNode (./node_modules/js-yaml/lib/loader.js:1442:11)
    at readBlockMapping (./node_modules/js-yaml/lib/loader.js:1104:12)
    at composeNode (./node_modules/js-yaml/lib/loader.js:1441:12)
    at readBlockMapping (./node_modules/js-yaml/lib/loader.js:1164:11)
    at composeNode (./node_modules/js-yaml/lib/loader.js:1441:12)
    at readBlockMapping (./node_modules/js-yaml/lib/loader.js:1164:11)
    at composeNode (./node_modules/js-yaml/lib/loader.js:1441:12)
    at readBlockMapping (./node_modules/js-yaml/lib/loader.js:1164:11)
    at composeNode (./node_modules/js-yaml/lib/loader.js:1441:12)
    at readDocument (./node_modules/js-yaml/lib/loader.js:1625:3)
    at loadDocuments (./node_modules/js-yaml/lib/loader.js:1688:5)
    at load (./node_modules/js-yaml/lib/loader.js:1714:19)
    at parseSingleYaml (./node_modules/renovate/lib/util/yaml.ts:63:29)
    at Object.extractPackageFile (./node_modules/renovate/lib/modules/manager/docker-compose/extract.ts:39:29)
    at extractPackageFile (./node_modules/renovate/lib/modules/manager/index.ts:75:9)"
}
```
## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by secustor — COMMENTED** [review]
Please add a test case for this.

**Review by viceice — DISMISSED** [review]

File: lib/modules/manager/docker-compose/extract.spec.ts [hunk 1]
```diff
@@ -58,11 +58,11 @@ describe('modules/manager/docker-compose/extract', () => {
     it('extracts image and replaces registry', () => {
       const res = extractPackageFile(
         `
-    version: "3"
-    services:
-      nginx:
-        image: quay.io/nginx:0.0.1
-      `,
```
Comment (viceice) [inline lib/modules/manager/docker-compose/extract.spec.ts:65]: use `codeBlock` string template helper

File: lib/modules/manager/docker-compose/extract.spec.ts [hunk 2]
```diff
@@ -145,5 +137,18 @@ describe('modules/manager/docker-compose/extract', () => {
...
+      const compose = codeBlock`
+        version: "3"
+        services:
+          nginx:
+            image: quay.io/nginx:0.0.1
+            envrionment:
+              {{ services['nginx']['env'] }}
+      `;
+      const res = extractPackageFile(compose, '', {});
+      expect(res?.deps).toHaveLength(1);
```
Comment (secustor) [inline lib/modules/manager/docker-compose/extract.spec.ts:151]:
```suggestion
      expect(res).toEqual({
```
Please use toEqual

**Review by viceice — COMMENTED** [review]
see open conversation

**Review by viceice — APPROVED** [review]

**Pull request merged**